File: Backend/src/modules/tooltip/material/spell_tooltip.rs
use crate::modules::tooltip::domain_value::SpellCost;

#[derive(Debug, Clone, Serialize, JsonSchema)]
pub struct SpellTooltip {
    pub name: String,
    pub icon: String,
    pub subtext: String,
    pub spell_cost: Option<SpellCost>,
    pub range: u32,
    pub description: String,
}

File: Webclient/src/app/module/tooltip/module/spell_tooltip/domain_value/spell_tooltip.ts
export interface SpellTooltip {
    spell_id: number;
    name: string;
    icon: string;
    subtext: string;
    spell_cost: {
        cost: number;
        cost_in_percent: boolean;
        power_type: string;
    } | null;
    range: number;
    description: string;
}
